tests/test_integration.py

```diff
@@ -582,6 +582,45 @@ def test_offline_migration_drbd(self):
 
         self.check_vm_present()
 
+    def test_offline_migration_xfs(self):
+        # Migrate without disk resizing
+        vm_migrate(
+            VM_HOSTNAME,
+            offline=True,
+            offline_transport='xfs',
+        )
+        self.check_vm_present()
+
+        disk_size_gib = self.vm_obj['disk_size_gib']
+
+        # Increase disk size during migration
+        vm_migrate(
+            VM_HOSTNAME,
+            offline=True,
+            offline_transport='xfs',
+            disk_size=disk_size_gib + 1,
+        )
+        self.check_vm_present()
+
+        # Decreasing disk size back during migration
```

Comment:
This should be a dedicated test.

Reply:
I would leave it as one test for disk changing since first it increases the disk and then decreases to the original size.

Reply:
This are two scenarios for the user. One might increase a disk size with IGVM but never decrease it and vice versa.

Additionally having many small test methods allows us to run tests in parallel and will speed up the test pipeline.

Reply:
I see the point. But the parallel execution has the overhead of VM deployment, and to be sure that the decrease operation will work, I have to increase the disk in advance.

If you insist, it's possible to apply `vm_set_disk_size_gib` before xfs migration with disk size decrease

Reply:
Split the test into two different